src/policy/semantic.rs

```diff
@@ -82,6 +82,78 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
             }
         }
     }
+
+    /// This function computes whether the current policy entails the second one.
+    /// A |- B means every sasifaction of A is also a satisfaction of B.
+    /// This implementation will run slow for larger policies but should be sufficient for
+    /// most practical policies.
+
+    // This algorithm has a naive implementation. It is possible to optimize this
+    // by memoizing and maintaining a hashmap.
+    pub fn entails(self, other: Policy<Pk>) -> bool {
```

Comment:
I think we should require the caller do the normalization step, and we should be able to take `other` by reference

Reply:
Ah I see, the algorithm itself does no normalization, so after setting things to trivial/unsatisfiable, we need to renormalize anyway.

Reply:
I'd be okay with this if you put a recursion depth counter in and refused to go further than 16 deep say. As written this function may run effectively forever and there is no good way for the user to detect or prevent this.

Reply:
A simpler way might be limit to the number of distinct witnesses as the depth is exactly equal to that. I think 24 should be good because we can support liquid as that is 15 + 3 = 18 keys

Reply:
Agreed, we should add a method to `Policy` which returns this. Call it `n_terminals` or something.

Reply:
The PR already has such a function `num_contraints`.